flax/linen/linear.py

```diff
@@ -302,26 +311,101 @@ def maybe_broadcast(x):
+      assert in_features % self.feature_group_count == 0
+      kernel_shape = kernel_size + (
+          in_features // self.feature_group_count, self.features)
+
+    else:
+      if self.feature_group_count != 1:
+        raise NotImplementedError(
+            f'`lax.conv_general_dilated_local` does not support '
+            f'`feature_group_count != 1`, got `{self.feature_group_count}`.'
+        )
+
+      # Need to know the spatial output shape of a standard convolution to
+      # create the unshared convolution kernel.
+      conv_output_shape = eval_shape(
+          lambda lhs, rhs: lax.conv_general_dilated(
+              lhs=lhs,
+              rhs=rhs,
+              window_strides=strides,
+              padding=padding_lax,
+              dimension_numbers=dimension_numbers
+          ),
+          inputs,
+          ShapedArray(kernel_size + (in_features, self.features), inputs.dtype)
+      ).shape
+
+      # One (unshared) convolutional kernel per each pixel in the output.
+      kernel_shape = conv_output_shape[1:-1] + (
+          np.prod(kernel_size) * in_features, self.features)
+
+    kernel = self.param('kernel', self.kernel_init, kernel_shape, self.param_dtype)
+    kernel = jnp.asarray(kernel, self.dtype)
+
+    if self.shared_weights:
+      y = lax.conv_general_dilated(
+          inputs,
+          kernel,
+          strides,
+          padding_lax,
+          lhs_dilation=input_dilation,
+          rhs_dilation=kernel_dilation,
+          dimension_numbers=dimension_numbers,
+          feature_group_count=self.feature_group_count,
+          precision=self.precision
+      )
+    else:
+      y = lax.conv_general_dilated_local(
+          lhs=inputs,
+          rhs=kernel,
+          window_strides=strides,
+          padding=padding_lax,
+          filter_shape=kernel_size,
+          lhs_dilation=input_dilation,
+          rhs_dilation=kernel_dilation,
+          dimension_numbers=dimension_numbers,
+          precision=self.precision
+      )
 
-    if is_single_input:
-      y = jnp.squeeze(y, axis=0)
     if self.use_bias:
-      bias = self.param('bias', self.bias_init, (self.features,), self.param_dtype)
+      if self.shared_weights:
+        # One bias weight per output channel, shared between pixels.
+        bias_shape = (self.features,)
+      else:
+        # One bias weight per output entry, unshared betwen pixels.
+        bias_shape = y.shape[1:]
+
+      bias = self.param('bias', self.bias_init, bias_shape, self.param_dtype)
       bias = jnp.asarray(bias, self.dtype)
-      y += jnp.reshape(bias, (1,) * (y.ndim - 1) + (-1,))
+      bias = bias.reshape((1,) * (y.ndim - bias.ndim) + bias.shape)
+      y += bias
+
+    if is_single_input:
+      y = jnp.squeeze(y, axis=0)
     return y
 
 
+class Conv(_Conv):
+  """Convolution Module wrapping `lax.conv_general_dilated`."""
+
+  @property
+  def shared_weights(self) -> bool:
+    return True
+
+
+class ConvLocal(_Conv):
```

Comment:
This one still needs to be exposed in docs/flax.linen.rst

Reply:
Thanks, done!